src/Questions.jsx

```diff
@@ -37,14 +39,23 @@ import './App.css'
            */
 const Questions = ({ questions, start, setStart }) => {
 
+    // You have quite a bit of state here
+    // It would be better if you could take your time to
+    // think what would be the best model to store each question into
+    // (hint: you are free to manipulate the incoming API data)
+
+
     /*  Answers state for storing all the shuffled answer*/
-    const [answers, setAnswers] = React.useState([]);
+    const [answers, setAnswers] = React.useState([]); // <--- this won't be needed once you got your questions right
     /*const [selectedAnswers, setSelectedAnswers] = React.useState({ qIndex: null, useranswer: null })*/
-    const [selectedAnswers, setSelectedAnswers] = React.useState(Array(questions.length).fill(null))
-    const [correctCount, setCorrectCount] = React.useState(0)
+    const [selectedAnswers, setSelectedAnswers] = React.useState(Array(questions.length).fill(null)) // <--- this won't be needed once you got your questions right
+    const [correctCount, setCorrectCount] = React.useState(0) // <--- this won't be needed once you got your questions right
     const [toggle, setToggle] = React.useState(false)
     const [restart, setRestart] = React.useState(false)
 
+    // useEffect is meant to mainly handle side-effect
+    // If you want to shuffle your answers, you should
+    // do that before storing them into your state
```

Comment:
here I have used use effect because every time when i am selecting any answer the selected answer state is getting changed and that's causing re-render and the answers are getting reshuffled.
t

Reply:
Alright, so from what I've understood, you want to reshuffle the answers when the user selects one?
That doesn't sound like a good UX? 🤔 

I understand your perspective but this shouldn't really be considered as a side effect IMO.
It would be better to do the shuffling when you receive the data + when you update the sate.